## **TASK 3** ##

A year's worth of sales from a fictitious pizza place, including the date and time of each order and the pizzas served, with additional details on the type, size, quantity, price and ingredients

In [1]:
import pandas as pd

In [2]:
orders = pd.read_csv("C:/Users/Joel/Documents/PYTHON/orders.csv")

orders.head()

order_id        date      time
0         1  2015-01-01  11:38:36
1         2  2015-01-01  11:57:40
2         3  2015-01-01  12:12:28
3         4  2015-01-01  12:16:31
4         5  2015-01-01  12:21:30

In [3]:
order_details = pd.read_csv("C:/Users/Joel/Documents/PYTHON/order_details.csv")

order_details.head()

order_details_id  order_id       pizza_id  quantity
0                 1         1     hawaiian_m         1
1                 2         2  classic_dlx_m         1
2                 3         2  five_cheese_l         1
3                 4         2    ital_supr_l         1
4                 5         2     mexicana_m         1

In [4]:
pizzas = pd.read_csv("C:/Users/Joel/Documents/PYTHON/pizzas.csv")

pizzas.head()

pizza_id pizza_type_id size  price
0   bbq_ckn_s       bbq_ckn    S  12.75
1   bbq_ckn_m       bbq_ckn    M  16.75
2   bbq_ckn_l       bbq_ckn    L  20.75
3  cali_ckn_s      cali_ckn    S  12.75
4  cali_ckn_m      cali_ckn    M  16.75

In [5]:
pizza_types = pd.read_csv("C:/Users/Joel/Documents/PYTHON/pizza_types.csv", encoding = "latin 1")

pizza_types.head()

pizza_type_id                          name category  \
0       bbq_ckn    The Barbecue Chicken Pizza  Chicken   
1      cali_ckn  The California Chicken Pizza  Chicken   
2   ckn_alfredo     The Chicken Alfredo Pizza  Chicken   
3     ckn_pesto       The Chicken Pesto Pizza  Chicken   
4    southw_ckn   The Southwest Chicken Pizza  Chicken   

                                         ingredients  
0  Barbecued Chicken, Red Peppers, Green Peppers,...  
1  Chicken, Artichoke, Spinach, Garlic, Jalapeno ...  
2  Chicken, Red Onions, Red Peppers, Mushrooms, A...  
3  Chicken, Tomatoes, Red Peppers, Spinach, Garli...  
4  Chicken, Tomatoes, Red Peppers, Red Onions, Ja...

In [6]:
df = pd.merge(order_details, orders, on="order_id")
df = pd.merge(df, pizzas, on = "pizza_id")
df = pd.merge(df, pizza_types, on="pizza_type_id")

In [12]:
df.head()

order_details_id  order_id       pizza_id  quantity        date      time  \
0                 1         1     hawaiian_m         1  2015-01-01  11:38:36   
1                 2         2  classic_dlx_m         1  2015-01-01  11:57:40   
2                 3         2  five_cheese_l         1  2015-01-01  11:57:40   
3                 4         2    ital_supr_l         1  2015-01-01  11:57:40   
4                 5         2     mexicana_m         1  2015-01-01  11:57:40   

  pizza_type_id size  price                       name category  \
0      hawaiian    M  13.25         The Hawaiian Pizza  Classic   
1   classic_dlx    M  16.00   The Classic Deluxe Pizza  Classic   
2   five_cheese    L  18.50      The Five Cheese Pizza   Veggie   
3     ital_supr    L  20.75  The Italian Supreme Pizza  Supreme   
4      mexicana    M  16.00         The Mexicana Pizza   Veggie   

                                         ingredients  
0           Sliced Ham, Pineapple, Mozzarella Cheese  
1  Pepperoni, Mushrooms, Red Onions, Red Peppers,...  
2  Mozzarella Cheese, Provolone Cheese, Smoked Go...  
3  Calabrese Salami, Capocollo, Tomatoes, Red Oni...  
4  Tomatoes, Red Peppers, Jalapeno Peppers, Red O...

In [13]:
df.columns

Index(['order_details_id', 'order_id', 'pizza_id', 'quantity', 'date', 'time',
       'pizza_type_id', 'size', 'price', 'name', 'category', 'ingredients'],
      dtype='object')

In [8]:
df.to_csv("PizzaPlaceSales_merged.csv", index=False)

In [39]:
df= pd.read_csv("PizzaPlaceSales_merged.csv")

In [49]:
#DATA CLEANING

#Converting of data
df["date"] = pd.to_datetime(df["date"], errors="coerce")
df["time"] = pd.to_datetime(df["time"], format="%H: %M: %S", errors="coerce").dt.time
df["price"] = pd.to_numeric(df["price"], errors= "coerce")
df["quantity"]= pd.to_numeric(df["quantity"], errors="coerce")

#Remove Duplicates
df = df.drop_duplicates()

#Standardizing text columns
df["name"] = df["name"].str.strip().str.lower()
df["category"] = df["category"].str.strip().str.lower()

#Creating Useful Colunms
df["sales"] = df["quantity"] * df["price"]
df["hour"] = pd.to_datetime(df["time"].astype(str), errors="coerce").dt.hour
df["day_of_week"] = df["date"].dt.day_name()
df["month"] = df["date"].dt.month_name()

#ANSWERING THE QUESTIONS
print("1. Total Revenue:", df["sales"].sum())
print("2. Total Quantity Sold:", df["quantity"].sum())
print("3. Total Orders:", df["order_id"].nunique())
print("4. Number of Pizza Types:", df["pizza_type_id"].nunique())
print("5. Average Pizza Price:", df["price"].mean())


peak_hours = (df.groupby("hour")["sales"].sum().sort_values(ascending=True))
print("6. Peak Hours pf Sales:", peak_hours.head())

print("7. Sales by Day of Week:")
print(df.groupby("day_of_week")["sales"].sum().sort_values(ascending=False))

print("8. Top 5 Bestselling Pizzas:")
print(df.groupby("name")["quantity"].sum().sort_values(ascending=False).head())

print("9. Monthly Sales Trend:")
print(df.groupby("month")["sales"].sum().sort_values(ascending=False))

print("10. Low Selling Pizza(Bottom 5):")
print(df.groupby("name")["quantity"].sum().sort_values().head(5))

1. Total Revenue: 817860.05
2. Total Quantity Sold: 49574
3. Total Orders: 21350
4. Number of Pizza Types: 32
5. Average Pizza Price: 16.494132044426163
6. Peak Hours pf Sales: Series([], Name: sales, dtype: float64)
7. Sales by Day of Week:
day_of_week
Friday       136073.90
Thursday     123528.50
Saturday     123182.40
Wednesday    114408.40
Tuesday      114133.80
Monday       107329.55
Sunday        99203.50
Name: sales, dtype: float64
8. Top 5 Bestselling Pizzas:
name
the classic deluxe pizza      2453
the barbecue chicken pizza    2432
the hawaiian pizza            2422
the pepperoni pizza           2418
the thai chicken pizza        2371
Name: quantity, dtype: int64
9. Monthly Sales Trend:
month
July         72557.90
May          71402.75
March        70397.10
November     70395.35
January      69793.30
April        68736.80
August       68278.25
June         68230.20
February     65159.60
December     64701.15
September    64180.05
October      64027.60
Name: sales, dtype: float

1) The Total Revenue is 817860.05
   
2) The Total Quantity Sold is 49574


3) The Total pizza ordered is 21350

   
4) The Number of Pizza Types sold is 32 types


5) The Average Pizza Price is 16.494132044426163


7) Here is the breakdown of total sales made on each day of the week
Sales by Day of Week:
Friday       136073.90
Thursday     123528.50
Saturday     123182.40
Wednesday    114408.40
Tuesday      114133.80
Monday       107329.55
Sunday        99203.50

Based on this, the highest sale here is 136073.90 which happened on Friday, so Friday is the day when sales are made the most

8) Here is the Top 5 Bestselling Pizzas:
name
The classic deluxe pizza      2453
The barbecue chicken pizza    2432
The hawaiian pizza            2422
The pepperoni pizza           2418
The thai chicken pizza        2371

9) Here is the Monthly Sales:
Month
July         72557.90
May          71402.75
March        70397.10
November     70395.35
January      69793.30
April        68736.80
August       68278.25
June         68230.20
February     65159.60
December     64701.15
September    64180.05
October      64027.60

A noticable trend here is that the sales isn't stable, it kept on fluctuating from one month to the other
in January we have 69793.30
in February we have 65159.60
in March we have 70395.35
in April we have 68736.80
in May we have 71402.75
in June we have 68230.20
in July we have 72557.90
in August we have 68278.25
in September we have 64180.05
in October we have 64027.60
in November we have 70395.35
in December we have 64701.15

10) Low Selling Pizza includes the following:

the brie carre pizza         490
the mediterranean pizza      934
the calabrese pizza          937
the spinach supreme pizza    950
the soppressata pizza        961